src/storage.rs

```diff
@@ -1103,6 +1103,26 @@ fn find_last_ball_mci_of_mci(db: &Connection, mci: u32) -> Result<u32> {
     Ok(rows[0].main_chain_index)
 }
 
+pub fn read_free_joints() -> Result<Vec<Joint>> {
+    let db = db::DB_POOL.get_connection();
+    let mut stmt = db.prepare_cached(
+        "SELECT units.unit FROM units LEFT JOIN archived_joints USING(unit) WHERE is_free=1 AND archived_joints.unit IS NULL",
+    )?;
+
+    let units = stmt.query_map(&[], |row| row.get::<_, String>(0))?;
+    let mut joints = Vec::new();
+    for unit in units {
+        if unit.is_err() {
```

Comment:
here we'd better use `match` if branch logic contains `return`, `break` or `continue` flow control statment